File: JavaScript/challenge_203.js
/*
  Challenge 203; STRINGS AND ARROWS

*/

var leftArrow = '<--<<';
var rightArrow = '>>-->';
function numArrows(s) {
  var matches = 0;
  for(var i = 0; i < s.length; i++) {
    var aSlice = s.slice(i,i+5);
    if (aSlice.length < 5 ) {
      break;
    }
    if ( aSlice === leftArrow || aSlice === rightArrow ) {
      matches++;
    }
  }
  return matches;
}

var fs  = require("fs");
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line !== "") { // ignore empty lines
        console.log(numArrows(line));
    }
});

File: Ruby/challenge_203.rb

myfile = ARGV[0]

if File.exists?(myfile)
  File.open(myfile).each_line do |line|
    line.strip!
    next if line.empty?
    _left_arrow = '<--<<'
    _right_arrow = '>>-->'
    matches = 0
    0.upto(line.size - 1) do |i|
      t = line.slice(i, 5)
      matches += 1 if t == _left_arrow || t == _right_arrow
    end
    puts matches
  end
end
